src/main/io/serial_cli.c

```diff
@@ -2143,6 +2166,81 @@ static void cliFeature(char *cmdline)
     }
 }
 
+#ifdef BEEPER
+static void cliBeeper(char *cmdline)
+{
+    uint32_t i;
+    uint32_t len = strlen(cmdline);;
+    uint8_t beeperCount = beeperTableEntryCount();
+    uint32_t mask = getBeeperOffMask();
+
+    if (len == 0) {
+        cliPrintf("Disabled:");
+        if (mask) {
+           for (int i = 0; i < beeperCount - 1; i++)
+               if (mask & (1 << i))
+                   cliPrintf("  %s", beeperNameForTableIndex(i));
+        } else {
+           cliPrint("  none");
+        }
+        return;
+    }
+
+    while (cmdline[0] == ' ') {
+        cmdline++;
+        len--;
+    }
+
+    if (strcasecmp(cmdline, "list") == 0) {
+        cliPrint("Available:");
+        for (i = 0; i < beeperCount; i++)
+            cliPrintf("  %s", beeperNameForTableIndex(i));
+        cliPrint("\r\n");
+        return;
+    } else {
+        bool remove = false;
+        if (cmdline[0] == '-') {
+            remove = true;     // this is for beeper OFF condition
+            cmdline++;
+            len--;
+        }
+        while (cmdline[0] == ' ') {
+            cmdline++;
+            len--;
+        }
+
+        for (i = 0; ; i++) {
```

Comment:
I'd prefer to untangle a bit using a for loop with endpoint and function return from within the loop if the beep was matched.